File: app/src/main/java/com/nptel/courses/online/utility/CustomBindingAdapter.java
package com.nptel.courses.online.utility;

import android.graphics.drawable.Drawable;
import android.widget.ImageView;

import androidx.appcompat.widget.AppCompatImageView;
import androidx.databinding.BindingAdapter;

import com.squareup.picasso.Picasso;

public class CustomBindingAdapter {

    @BindingAdapter("app:url")
    public static void setUrl(ImageView imageView, String url) {
        Picasso.get().load(url).into(imageView);
    }

    @BindingAdapter("app:srcCompat")
    public static void setSrcCompat(AppCompatImageView imageView, Drawable drawable) {
        imageView.setImageDrawable(drawable);
    }
}

File: app/src/main/java/com/nptel/courses/online/utility/CustomBindingAdapter.kt
package com.nptel.courses.online.utility

import android.graphics.drawable.Drawable
import android.widget.ImageView
import androidx.appcompat.widget.AppCompatImageView
import androidx.databinding.BindingAdapter
import com.squareup.picasso.Picasso

object CustomBindingAdapter {
    @JvmStatic
    @BindingAdapter("app:url")
    fun setUrl(imageView: ImageView?, url: String?) {
        Picasso.get().load(url).into(imageView)
    }

    @BindingAdapter("app:srcCompat")
    fun setSrcCompat(imageView: AppCompatImageView, drawable: Drawable?) {
        imageView.setImageDrawable(drawable)
    }
}